File: ios/UniPark/UniPark/domain/repositories/ViolationRepository.swift
import Foundation

// TODO: implemented in data/repositories/
public protocol ViolationRepository {
	func reportViolation(vehicleId: UUID?, lotId: UUID?, reason: String, evidenceUrl: String?) async throws -> Violation
	func fetchViolations(status: ViolationStatus?) async throws -> [Violation]
	func resolveViolation(id: UUID, status: ViolationStatus) async throws -> Violation
}

File: android/app/src/main/java/com/unipark/android/domain/repositories/ViolationRepository.kt
package com.unipark.android.domain.repositories

import com.unipark.android.domain.entities.Violation
import java.util.UUID

interface ViolationRepository {
    suspend fun reportViolation(
        vehicleId: UUID,
        lotId: UUID,
        reason: String,
        photoUri: String?,
    ): Violation
}
